src/components/InactivityHandler.tsx

```diff
@@ -0,0 +1,155 @@
+ */
+export class InactivityHandler extends React.Component<InactivityHandlerProps, InactivityHandlerState> {
+    public static defaultProps = {
+        dimBrightness: DEFAULT_DIM_BRIGHTNESS,
+        idleTime: DEFAULT_IDLE_TIME
+    };
+
+    private _panResponder: PanResponderInstance;
+    private idleTimer: any;
+
+    public constructor(props: InactivityHandlerProps) {
+        super(props);
+    }
+
+    public componentWillMount(): void {
+        this.setState({ active: true, currentState: "" });
+
+        this.props.navigation.addListener("didFocus", () => this.componentDidFocus());
+        this.props.navigation.addListener("didBlur", () => this.componentDidBlur());
+
+        AppState.addEventListener("change", this.handleAppStateChange.bind(this));
+
+        this._panResponder = PanResponder.create({
+            onMoveShouldSetPanResponder: () => false,
+            onStartShouldSetPanResponder: () => false,
+
+            onMoveShouldSetPanResponderCapture: this.handlePanResponderCapture.bind(this),
+            onStartShouldSetPanResponderCapture: this.handlePanResponderCapture.bind(this)
+        });
+    }
+
+    public componentWillUnmount(): void {
+        AppState.removeEventListener("change", this.handleAppStateChange.bind(this));
+    }
+
+    public render(): ReactNode {
+        return (
+            <View style={{ flex: 1 }} {...this._panResponder.panHandlers} >
+                <TouchableWithoutFeedback style={{ flex: 1 }} >
+                    <View style={{ flex: 1 }} >
+                        {this.props.children}
+                    </View>
+                </TouchableWithoutFeedback>
+            </View>
+        );
+    }
+
+    /**
+     * Screen focused. Resets timer.
+     */
+    public componentDidFocus(): void {
+        clearTimeout(this.idleTimer);
+        this.setIdleTimer();
+    }
+
+    /**
+     * Screen loses focus. Removes timer.
+     */
+    public componentDidBlur(): void {
+        clearTimeout(this.idleTimer);
+        this.onActiveChange(true);
+    }
+
+    /**
+     * If movement is detected, active is set to true and idleTimer is set.
+     */
+    public handlePanResponderCapture(): boolean {
+        clearTimeout(this.idleTimer);
+
+        // set as active
+        if (!this.state.active) {
+            this.setState({ active: true });
+            this.onActiveChange(true);
+        }
+
+        // set idle timer
+        this.setIdleTimer();
+
+        return false;
+    }
+
+    /**
+     * Sets idle timer. When timer stops, active is set to false.
+     */
+    public setIdleTimer(): void {
+        this.idleTimer = setTimeout(() => {
+            this.setState({ active: false });
+            this.onActiveChange(false);
+        }, this.props.idleTime);
+    }
+
+    /**
+     * Called to handle active change.
+     * If active, set brightness to system brightness.
+     * If inactive, dim brightness to 0.
+     */
+    public onActiveChange(active: boolean): void {
```

Comment:
After click-through testing, I found that in iOS it looks like Expo's `Brightness.setBrightnessAsync(...)` directly modifies the system brightness. I'm not sure if it does the same on Android, but the effect is that:

- Dimming on inactivity directly modifies the system brightness
- The subsequent call to `Brightness.getSystemBrightnessAsync()` returns the already-dimmed brightness, meaning that...
- Returning the brightness to current system brightness on activity has no effect

My proposed solution is to grab the system brightness in `componentWillMount()` and hold onto it as the "previous" system brightness.

Reply:
Yeah, it seems like it works a bit differently in iOS. I like the proposed solution, will add that later today